File: Domain/LineaDeVenta.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain
{
    public class LineaDeVenta
    {
        private Stock _stock;
        private int _cantidad;

        public LineaDeVenta(Stock stock, int cantidad)
        {
            _stock = stock;
            if (cantidad > stock.stock)
            {
                throw new ArgumentException("No hay stock suficiente");
            }
            _cantidad = cantidad;
        }

        public double GetPrecio()
        {
            double precioUnitario = _stock.articulo.precio;
            double precio = precioUnitario * _cantidad;
            return precio;
        }

    }
}

File: src/main/java/is/tp3/domain/LineaDeVenta.java
package is.tp3.domain;

public class LineaDeVenta {

    Articulo articulo;

    Integer cantidad;


    public LineaDeVenta( ){

    }


    public void agregarArticuloALineaDeVenta(Integer cantidad,Articulo articulo) {
        this.articulo = articulo;
        this.cantidad = cantidad;


    }

    public int getCantidadArticulos() {
        return this.cantidad;
    }

    public Articulo getArticulo() {
        return this.articulo;
    }

    public double getSubTotal() {
        return cantidad * articulo.getPrecio();
    }
}
